### PR TITLE
Feature/#6 kitchen view - Retrieve orders

### PR DESCRIPTION
**Pull request by chloe-hodgson**
Displaying orders as JSON string for now - still need to make it look pretty and add a method to the order service to update the status 

I've created a shared folder to keep the order service and models, as I imagine it will be used by a few components (waiters can add orders, kitchen needs to view orders etc.).

### CODE REVIEW

File: client/app/shared/order.service.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    getOrders(): Observable<any> {
+        return this.http.get("/api/orders/in-progress")
+            .pipe(
+                map((response: any) => this.toOrderArray(response)),
+                catchError((error: any) => observableThrowError(error))
+            );
+    }
+
+
+    toOrderArray(response: any): Order[] {
```
Comment (dorusomcutean) [inline client/app/shared/order.service.ts]: I'm not sure this function is necessary here. getOrders() already gives you the response and you don't usually have functions like this in the service. 

The service usually only has functions that call the API. You can use the model in getOrders(), like this: 
`map((response: Order) => OrderFactory.create(order)` if you want to map it and have it strictly typed but we don't really need to do it this way I think... 

The tutor will probably not even look into these details... 

@last1here @jessicawong217 @Nadiajayy What do you guys think?

File: routes/orders/order.controller.js [hunk 2]
```diff
@@ -13,6 +13,18 @@ function list(req, res, next) {
...
 
+/**
+ * Return a list of incomplete orders.
+ * @param {*} req The express request object.
+ * @param {*} res The express result object.
+ * @param {*} next Next match route handler.
+ */
+function listInProgress(req, res, next) {
+    Order.get("InProgress")
+        .then(users => res.json(users))
```
Comment (dorusomcutean) [inline routes/orders/order.controller.js]: Users can be changed to data or items I think, something more specific to this context. I need to change it this in the counter functions as well...

Comment (chloe-hodgson) [reply routes/orders/order.controller.js]: Ah yeah I didn't spot that - copy and paste overlook!

File: client/app/shared/order.service.ts (hunk 1 above)
Comment (chloe-hodgson) [reply client/app/shared/order.service.ts]: To be honest I've just copied how we do it in work! Happy to change it if it's not necessary :)

Comment (last1here) [reply client/app/shared/order.service.ts]: Could also cast it within the function  if we dont plan to add methods to the Order `return this.http.get("/api/orders/in-progress") as Observable<Order[]>` or within the pipe. Or just set the return type of the function `getOrders(): Observable<Order[]>` saves the extra loop and assignments. 

Would need to change the types `id` to `_id` to match the response.

File: routes/orders/order.model.js [hunk 3]
```diff
@@ -47,12 +44,18 @@ OrderSchema.statics = {
...
+    list({skip = 0, limit = 50} = {}) {
         return this.find()
-            .sort({ createdAt: -1 })
+            .sort({createdAt: -1})
             .skip(+skip)
             .limit(+limit)
             .exec();
+    },
+
+    get(orderStatus) {
```
Comment (last1here) [inline routes/orders/order.model.js]: Just realised this method overrides the previous `get(id)`, probably want to rename either or both to `getBy*` or something.

**Pull request merged**